iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -792,10 +798,9 @@ class expected<ValueType, ErrorType>
 
   private:
     expected(variant<ValueType, ErrorType>&& f_store, const bool hasError) noexcept;
-
-  private:
     variant<ValueType, ErrorType> m_store;
     bool m_hasError;
+    bool m_isInitialized{true};
```

Comment:
This is ok if it is always ensured that any ctor constructs a valid object or sets it to false otherwise. Otherwise we need to default it to false and set to true in ctor.